ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/max/MaxComparableKudaf.java

```diff
@@ -15,22 +15,23 @@
 
 package io.confluent.ksql.function.udaf.max;
 
-import io.confluent.ksql.function.udaf.BaseNumberKudaf;
+import io.confluent.ksql.function.udaf.BaseComparableKudaf;
 import io.confluent.ksql.schema.ksql.types.SqlType;
-import java.math.BigDecimal;
+import io.confluent.ksql.schema.utils.FormatOptions;
 
-public class DecimalMaxKudaf extends BaseNumberKudaf<BigDecimal> {
+public class MaxComparableKudaf<T extends Comparable<? super T>> extends BaseComparableKudaf<T> {
```

Comment:
Same comment from BaseComparableKudaf. Should it reflect the intention of the class instead? Such as MaxNumberAndTimeTypesKudaf? Same comment for the Min. What do you think?

Reply:
Hey @spena thanks for chipping in!
[Line 47 
](https://github.com/confluentinc/ksql/pull/8924/files#diff-1c7a573f0ba0b7096f72b8355fcd3f4ba8fbe669aa2f5ed7df3c608df5b34945L47) should be changed indeed but I would go for something generic like "comparable".
With the latest refactoring the only thing we actually need to support Min/Max aggregation Kudaf is a comparable Type -- currently Number and Date/Time -- but this could change in the future right?

Reply:
Ooo.... you two are nerd sniping me this morning...  

If we are going to be precise about the names here, I think we can safely name this `MaxKudaf`.  The name of the other class is completely separately.  I think its signature just gives one what you'd want for a Fold/Reduce.  (E.g., see this issue: https://github.com/confluentinc/ksql/issues/8798)

Ok, so how can I justify dropping `Comparable` from the name? 

As a brief argument, (I think) it is easy to see that a Java `Comparable` type has a total ordering, and as such, it makes sense to talk about `Max` and `Min` values.  

Going the other way, if one has a `Max` (or a `Min`) function, then one could use it pairwise to write an implementation of `Comparable`.  (It'd look something like: {If `a==b`, return 0; else, if Max(a,b)==a, return 1, else return -1} (Up to a sign:)).)

At that point, the only thing which'd go wrong is someone having a type which is "comparable" which does not implement `Comparable`.  We'd have to have user defined types and then folks would have to work around that by adding a `Comparable` wrapper or they'd have to implement a UDAF that handles their "comparable" interface.  That'd be pretty rare...  

So where does that leave us?  I'm kinda ok with the name as-is (it doesn't hurt anything) or a smaller name (`MaxKudaf`).

Reply:
Thanks for the input here!
Introduced type `T` bounds and opted for simpler Min/Max Kudaf names in https://github.com/confluentinc/ksql/commit/2cb7546fc07f9285e3089fb916ad52b020332c92

Reply:
Awesome.  @spena any more thoughts?